GLAA.Services/PublicRegister/PublicRegisterViewModelBuilder.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using AutoMapper;
+using GLAA.Domain.Models;
+using GLAA.Repository;
+using GLAA.ViewModels.PublicRegister;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace GLAA.Services.PublicRegister
+{
+    public class PublicRegisterViewModelBuilder : IPublicRegisterViewModelBuilder
+    {
+        private readonly IEntityFrameworkRepository repository;
+        private readonly ILicenceRepository licenceRepository;
+        private readonly IMapper mapper;
+
+        private List<SelectListItem> _ukCountries;
+        public List<SelectListItem> UkCountries => new List<SelectListItem>(_ukCountries);
+
+        public PublicRegisterViewModelBuilder(
+            IMapper mapper,
+            IEntityFrameworkRepository repository,
+            ILicenceRepository licenceRepository)
+        {
+            this.mapper = mapper;
+            this.repository = repository;
+            this.licenceRepository = licenceRepository;
+
+            _ukCountries = repository.GetAll<Country>().Select(x =>
+                new SelectListItem { Value = x.Name, Text = x.Name }).ToList();
+        }
+
+        public PublicRegisterLicenceListViewModel BuildAllLicences()
+        {
+            return new PublicRegisterLicenceListViewModel
+            {
+                Title = "Public Register",
+                //We only want licences with status that are allowed to be shown using the "ShowInPublicRegister" field.
+                Licences = licenceRepository.GetAllLicences()
+                    .Where(x => GetLatestStatus(x).Status.ShowInPublicRegister)
+                    .Select(BuildSummary),
+                PublicRegisterSearchViewModel = BuildPublicRegisterSearchViewModel()
+            };
+        }
+
+        public PublicRegisterLicenceListViewModel BuildEmptySearch()
+        {
+            return new PublicRegisterLicenceListViewModel
+            {
+                Title = "Public Register",
```

Comment:
Does the title ever change? I think I'd be happy for it to be static in the view as it's (marginally) extra data getting sent about.

Reply:
Title removed.